src/mongocxx/test/spec/unified_test_format.cpp

```diff
@@ -233,11 +233,122 @@ entity::map& get_entity_map() {
     return m;
 }
 
+write_concern get_write_concern(const document::element& opts) {
+    if (!opts["writeConcern"])
+        return {};
+
+    auto wc = write_concern{};
+    if (auto w = opts["writeConcern"]["w"]) {
+        REQUIRE(w.type() == type::k_int32);  // TODO: support type k_utf8
+        wc.nodes(w.get_int32());
+    }
+
+    return wc;
+}
+
+read_concern get_read_concern(const document::element& opts) {
+    if (!opts["readConcern"])
+        return {};
+
+    auto rc = read_concern{};
+    if (auto level = opts["readConcern"]["level"])
+        rc.acknowledge_string(level.get_string().value);
+    return rc;
+}
+
+template <typename T>
+void set_common_options(T& t, const document::element& opts) {
+    if (!opts)
+        return;
+
+    t.read_concern(get_read_concern(opts));
+    t.write_concern(get_write_concern(opts));
+    REQUIRE_FALSE(/* TODO */ opts["readPreference"]);
+}
+
+options::gridfs::bucket get_bucket_options(document::view object) {
+    if (!object["bucketOptions"])
+        return {};
+
+    auto opts = options::gridfs::bucket{};
+    set_common_options(opts, object["bucketOptions"]);
+
+    if (auto name = object["bucketOptions"]["bucketName"])
+        opts.bucket_name(name.get_string().value.to_string());
+    if (auto size = object["bucketOptions"]["chunkSizeBytes"])
+        opts.chunk_size_bytes(size.get_int32().value);
+    REQUIRE_FALSE(/* TODO */ object["bucketOptions"]["disableMD5"]);
```

Comment:
I think the TODO can be removed. Drivers were only required to support `disableMD5` if they had a pre-existing GridFS implementation that used MD5. The `options::gridfs::bucket` class does not have a `disableMD5` option. Tests shouldn't be specifying it. And if they did, we cannot do anything with it, so throwing seems reasonable.